File: editor/en/[872]Leaf-Similar Trees.java

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public boolean leafSimilar(TreeNode root1, TreeNode root2) {
        List<Integer> list1 = new ArrayList<>();
        List<Integer> list2 = new ArrayList<>();
        getList(root1, list1);
        getList(root2, list2);
        return list1.equals(list2);
    }

    private void getList(TreeNode node, List<Integer> list) {
        if (node == null) {
            return;
        }

        if (node.left == null && node.right == null) {
            list.add(node.val);
        }

        getList(node.left, list);
        getList(node.right, list);
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[872]Leaf-Similar Trees.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root1
 * @param {TreeNode} root2
 * @return {boolean}
 */
var leafSimilar = function (root1, root2) {
    const list1 = [];
    const list2 = [];
    dfs(root1, list1);
    dfs(root2, list2);
    return isEqual(list1, list2);
};

function dfs(root, list) {
    if (root === null) {
        return;
    }

    if (!root.left && !root.right) {
        list.push(root.val);
    }

    dfs(root.left, list);
    dfs(root.right, list);
}

function isEqual(arr1, arr2) {
    if (arr1.length !== arr2.length) {
        return false;
    }

    for (let i = 0; i < arr1.length; i++) {
        if (arr1[i] !== arr2[i]) {
            return false;
        }
    }

    return true;
}

//leetcode submit region end(Prohibit modification and deletion)
